lib/stats.rb

```diff
@@ -1,42 +1,37 @@
-class Stats
-
-  def self.fetch(*opts)
-    Stats.new(opts)
+module Statistics
+ 
+  def all
+    {
+      :two_plus_two =>     Proc.new { 2 + 2 },
+      :four_plus_four =>   Proc.new { 4 + 4 },
+      :three_times_two =>  Proc.new { 3 * 2 }
+    }
   end
 
-  def initialize(opts)
-    @stats = opts
-    run_stats
-  end
+  class Collection
 
-  def run_stats
-    @stats.any? ? run_selected_stats : run_all_stats
-  end
+    extend Statistics
 
-  def run_selected_stats
-    @stats.each { |stat| send("set_#{stat}".to_sym) }
-  end
+    all.keys.each { |stat| self.class_eval { attr_reader stat } }
 
-  def run_all_stats
-    setter_methods = methods.grep /set_/
-    setter_methods.each do |method|
-      send(method)
+    def self.fetch(*opts)
+      Statistics::Collection.new(opts)
     end
-  end
 
-  attr_reader :stat_one
-  attr_reader :stat_two
-  attr_reader :stat_three
+    def initialize(opts)
+      run_stats(opts)
+    end
 
-  def set_stat_one
-    @stat_one = 1000
-  end
+    def run_stats(opts)
+      opts.any? ? run_selected_stats(opts) : run_all_stats
+    end
 
-  def set_stat_two
-    @stat_two = 2000
-  end
+    def run_selected_stats(opts)
+      opts.each { |stat| instance_variable_set("@#{stat}", self.class.all[stat].call)  }
+    end
 
-  def set_stat_three
-    @stat_three = 3000
+    def run_all_stats
+      self.class.all.each { |stat,proc|  instance_variable_set("@#{stat}", proc.call) }
```

Comment:
Line is too long. [87/80]<br>Unnecessary spacing detected.<br>Space missing after comma.